temba/flows/models.py

```diff
@@ -4412,6 +4441,68 @@ def execute(self, run, actionset_uuid, msg, offline_on=None):
         return []
 
 
+class UssdAction(ReplyAction):
+    """
+    USSD action to send outgoing USSD messages
+    Created from a USSD ruleset
+    It builds localised text with localised USSD menu support
+    """
+    TYPE = 'ussd'
+    MESSAGE = 'ussd_message'
+    MENU = 'ussd_menu'
+    TYPE_WAIT_USSD_MENU = 'wait_menu'
+    TYPE_WAIT_USSD = 'wait_ussd'
+
+    def __init__(self, msg=None, base_language=None, languages=None):
+        super(UssdAction, self).__init__(msg)
+        self.base_language = base_language
+        self.languages = languages
+
+    @staticmethod
+    def from_ruleset(rule, org):
+        if rule and hasattr(rule, 'config') and isinstance(rule.config, basestring):
+            # initial message, menu obj
+            obj = json.loads(rule.config)
+            msg = obj.get(UssdAction.MESSAGE, '')
+
+            # define language
+            base_language = org.primary_language.iso_code if org.primary_language else 'base'
+            org_languages = {l.iso_code for l in org.languages.all()}
+
+            # initialize UssdAction
+            ussd_action = UssdAction(msg, base_language, org_languages)
+
+            ussd_action.prepare_localised_msg()
+
+            if rule.ruleset_type == UssdAction.TYPE_WAIT_USSD_MENU:
+                ussd_action.add_menu_to_msg(obj)
+
+            return ussd_action
+        else:
+            return UssdAction()
+
+    def prepare_localised_msg(self):
+        # if there is a translation missing fill it with the base language
+        for language in self.languages:
+            if language not in self.msg:
+                self.msg[language] = self.msg[self.base_language]
```

Comment:
Sorry, as I recall, the issue wasn't precedence. In my test I believe it was blowing up following those repro steps because the flow didn't have a value for the org language.